pkg/controller/workload/workload_event.go

```diff
@@ -139,6 +136,8 @@ func removeWorkloadResource(removed_resources []string) error {
 	)
 
 	for _, workloadUid := range removed_resources {
+		go deleteWorkloadCache(workloadUid)
```

Comment:
Donot startup so many goroutines to do light weight work. It does not  improve the perf.

And you donot have a lock, it may panic

Reply:
fixed